packages/astro/src/core/app/index.ts

```diff
@@ -204,17 +222,27 @@ export class App {
 		page: SinglePageBuiltModule,
 		status = 200
 	): Promise<RenderContext> {
+		let preferredLocale: undefined | string = undefined;
+		let preferredLocaleList: undefined | string[] = undefined;
+		if (this.#manifest.i18n) {
+			const result = computePreferredLocales(request, this.#manifest.i18n.locales);
```

Comment:
As we talked about on the call, it would be good to compute these lazily only when the user asks for it. First, this will improve performance as we won't need to parse this in sites not using it, but also the warning we get in SSG will not occur.

To do this you'll need to pass the `locales` list down to here: https://github.com/withastro/astro/blob/main/packages/astro/src/core/render/result.ts#L187

That line is where cookies are parsed lazily. So something like this will be needed.